examples/recording_service/pluggable_storage/c/CMakeLists.txt

```diff
@@ -0,0 +1,135 @@
+#
+# (c) 2019 Copyright, Real-Time Innovations, Inc.  All rights reserved.
+#
+#  RTI grants Licensee a license to use, modify, compile, and create derivative
+#  works of the Software.  Licensee has the right to distribute object form
+#  only for use with RTI products.  The Software is provided "as is", with no
+#  warranty of any type, including any warranty for fitness for any purpose.
+#  RTI is under no obligation to maintain or support the Software.  RTI shall
+#  not be liable for any incidental or consequential damages arising out of the
+#  use or inability to use the software.
+#
+cmake_minimum_required(VERSION 3.11)
+
+project (PluggableStorage)
+
+# Find RTI Connext dependencies
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CONNEXTDDS_DIR}/resource/cmake")
+
+find_package(
+    RTIConnextDDS "6.0.0"
+    REQUIRED
+    COMPONENTS
+        core
+        routing_service
+)
+
+find_package(
+    RTICodeGenerator "3.0.0"
+    REQUIRED
+)
+
+set(GENERATED_CODE
+    "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsg.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsgPlugin.c"
+    "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsgSupport.c"
+)
+
+# Generate type plug-in files for the storage type
+add_custom_command(
+    OUTPUT ${GENERATED_CODE}
+    MAIN_DEPENDENCY "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsg.idl"
+    COMMAND ${RTICODEGEN} -language C -update typefiles
+            -platform ${CONNEXTDDS_ARCH} -d "${CMAKE_CURRENT_SOURCE_DIR}" 
+            "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsg.idl"
+)
+
+add_custom_target(
+    generated_code DEPENDS ${GENERATED_CODE}
+)
+
+# Define the library that will provide the storage writer plugin
+add_library(
+    FileStorageWriter MODULE
+        "${CMAKE_CURRENT_SOURCE_DIR}/HelloMsg.c" 
+        "${CMAKE_CURRENT_SOURCE_DIR}/FileStorageWriter.c"
+        "${CMAKE_CURRENT_SOURCE_DIR}/FileStorageWriter.h"
+)
+
+add_dependencies(FileStorageWriter generated_code)
+
+target_link_libraries(
+    FileStorageWriter
+    RTIConnextDDS::routing_service_infrastructure
+    RTIConnextDDS::c_api
+    ${CONNEXTDDS_EXTERNAL_LIBS}
+)
+
+target_include_directories(
+    FileStorageWriter
+    PUBLIC "${CMAKE_BINARY_DIR}"
```

Comment:
Use ``CMAKE_CURRENT_BINARY_DIR`` instead. If somebody includes this project as sub-project, the build will fail.